src/tike/ptycho/ptycho.py

```diff
@@ -318,13 +323,17 @@ def _rescale_obj_probe(operator, pool, data, psi, scan, probe):
     return probe
 
 
-def split_by_scan_grid(order, data, scan, shape, fly=1):
+def split_by_scan_grid(rank, size, order, data, scan, shape, fly=1):
```

Comment:
This function does not need to know about rank and size. We can choose a `shape` that is `processes * threads`. Then split the output of `split_by_scan_grid`.

Reply:
Logically, it's better to let the split function aware of the processes, so each process only needs to handle its own trunk of data. Otherwise, every process should redundantly split the entire dataset.